rpcs3/Emu/System.cpp

```diff
@@ -3460,7 +3460,7 @@ utils::serial* Emulator::DeserialManager() const
 
 bool Emulator::IsVsh()
 {
-	return g_ps3_process_info.get_cellos_appname() == "vsh.self"sv;
+	return g_ps3_process_info.self_info.prog_id_hdr.program_authority_id >> 52 == 0x107; // Not only VSH but also most CoreOS LV2 SELFs need the special treatment
```

Comment:
This getter was meant for vsh.self (as you can probably guess from the name).
I don't see how this random unexplained magic number combination can possibly fit all applications of IsVsh() in the codebase.
If anything this is confusing to say the least.

Reply:
During my tests with some CoreOS LV2 SELFs (like ps3swu.self and emer_init.self), mostly they turn out to need the same special treatment as VSH needs, and those SELFs all have the PAID starting with 0x107xxxxxxxxxxxxx.
Reference: https://www.psdevwiki.com/ps3/Program_Authority_ID#Structure